apis/v1alpha2/gateway_types.go

```diff
@@ -663,19 +651,24 @@ const (
 
 // ListenerStatus is the status associated with a Listener.
 type ListenerStatus struct {
-	// Port is the unique Listener port value for which this message is
-	// reporting the status.
-	Port PortNumber `json:"port"`
-
-	// Protocol is the Listener protocol value for which this message is
-	// reporting the status.
-	Protocol ProtocolType `json:"protocol"`
+	// Name is the name of the Listener.
+	//
+	// +kubebuilder:validation:MaxLength=253
+	Name string `json:"name,omitempty"`
 
-	// Hostname is the Listener hostname value for which this message is
-	// reporting the status.
+	// SupportedKinds is the list indicating the Kinds supported by this
+	// listener. When this is not specified on the Listener, this MUST represent
+	// the kinds an implementation supports for the specified protocol. When
+	// there are kinds specified on the Listener, this MUST represent the
+	// intersection of those kinds and the kinds supported by the implementation
+	// for the specified protocol.
 	//
-	// +optional
-	Hostname *Hostname `json:"hostname,omitempty"`
+	// +kubebuilder:validation:MaxItems=8
```

Comment:
```suggestion
	// +kubebuilder:validation:MaxItems=16
```

Reply:
At present we only have 4 valid Route kinds, and it's unlikely/impossible that they'd all be valid for a given listener. We can always increase this limit in the future if needed, but I'd be hesitant to start too high here.